tests/test_pipeline/test_mixins.py

```diff
@@ -0,0 +1,205 @@
+from copy import deepcopy
+from unittest.mock import MagicMock
+
+import pandas as pd
+import pytest
+
+from etna.models.base import NonPredictionIntervalContextIgnorantAbstractModel
+from etna.models.base import NonPredictionIntervalContextRequiredAbstractModel
+from etna.models.base import PredictionIntervalContextIgnorantAbstractModel
+from etna.models.base import PredictionIntervalContextRequiredAbstractModel
+from etna.pipeline.mixins import PipelineModelPredictMixin
+from etna.transforms import DateFlagsTransform
+from etna.transforms import FilterFeaturesTransform
+
+
+def make_mixin(ts=None, model=None, transforms=(), mock_recreate_ts=True, mock_determine_prediction_size=True):
+    if ts is None:
+        ts = MagicMock()
+    if model is None:
+        model = MagicMock(spec=NonPredictionIntervalContextIgnorantAbstractModel)
+
+    mixin = PipelineModelPredictMixin()
+    mixin.ts = ts
+    mixin.transforms = transforms
+    mixin.model = model
+    if mock_recreate_ts:
+        mixin._recreate_ts = MagicMock()
+    if mock_determine_prediction_size:
+        mixin._determine_prediction_size = MagicMock()
+    return mixin
+
+
+@pytest.mark.parametrize("context_size", [0, 3])
+@pytest.mark.parametrize("start_idx, end_idx", [(0, 0), (0, 1), (0, 10), (5, 10)])
+@pytest.mark.parametrize(
+    "transforms",
+    [
+        [DateFlagsTransform()],
+        [FilterFeaturesTransform(exclude=["regressor_exog_weekend"])],
+        [DateFlagsTransform(), FilterFeaturesTransform(exclude=["regressor_exog_weekend"])],
+    ],
+)
+def test_prepare_ts(context_size, start_idx, end_idx, transforms, example_reg_tsds):
```

Comment:
It would be more obvious to specify expected values by hand: now expected values computed in runtime -- I guess it's a bit error prone solution: you use the same code to compute expected values - it helps to find bugs in regression testing but code could already has logical errors.

Reply:
Do you suggest to use `start_timestamp` and `end_timestamp` here instead of `start_idx` and `end_idx`?

Reply:
Or add `expected_start_idx` here?

Reply:
> start_timestamp and end_timestamp
Yep
I guess we know them without any computation